lib/Pipes/Yield/Assembly.cpp

```diff
@@ -0,0 +1,91 @@
+//
+// This file is distributed under the MIT License. See LICENSE.md for details.
+//
+
+#include "revng/EarlyFunctionAnalysis/FunctionMetadata.h"
+#include "revng/EarlyFunctionAnalysis/IRHelpers.h"
+#include "revng/Lift/LoadBinaryPass.h"
+#include "revng/Model/Binary.h"
+#include "revng/Pipeline/Pipe.h"
+#include "revng/Pipeline/RegisterContainerFactory.h"
+#include "revng/Pipeline/RegisterPipe.h"
+#include "revng/Pipes/Kinds.h"
+#include "revng/Pipes/ModelGlobal.h"
+#include "revng/Pipes/Yield/Assembly.h"
+#include "revng/Yield/Assembly/Assembly.h"
+#include "revng/Yield/Assembly/DisassemblyHelper.h"
+#include "revng/Yield/Assembly/HTML.h"
+
+namespace revng::pipes {
+
+void YieldAssemblyPipe::run(pipeline::Context &Context,
+                            const FileContainer &SourceBinary,
+                            const pipeline::LLVMContainer &TargetList,
+                            StringMapContainer &Output) {
+  // Access the model
+  const auto &Model = revng::pipes::getModelFromContext(Context);
+
+  // Access the binary
+  revng_assert(SourceBinary.path().has_value());
+  auto MaybeBinary = loadBinary(*Model, *SourceBinary.path());
+  revng_assert(MaybeBinary);
+  const RawBinaryView &BinaryView = MaybeBinary->first;
+
+  // Access the llvm module
+  const llvm::Module &Module = TargetList.getModule();
+
+  // Define the helper object to store the disassembly pipeline.
+  // This allows it to only be created once.
+  DissassemblyHelper Helper;
+
+  for (const auto &LLVMFunction : FunctionTags::Isolated.functions(&Module)) {
+    auto Metadata = extractFunctionMetadata(&LLVMFunction);
+    auto ModelFunctionIterator = Model->Functions.find(Metadata->Entry);
+    revng_assert(ModelFunctionIterator != Model->Functions.end());
+    const auto &Function = *ModelFunctionIterator;
+
+    auto Disassembled = Helper.disassemble(Function, *Metadata, BinaryView);
+
+    auto HTML = html::yield(Disassembled, *Metadata, *Model);
+    Output.insert_or_assign(Function.Entry, std::move(HTML));
+  }
+}
+
+void YieldAssemblyPipe::print(const pipeline::Context &,
+                              llvm::raw_ostream &OS,
+                              llvm::ArrayRef<std::string>) const {
+  OS << *revng::ResourceFinder.findFile("bin/revng");
+  OS << " ma-a-a-agic ^_^\n";
```

Comment:
@aleclearmind, BTW, this still needs to be fixed. Do you have any suggestions about how the command should look like?

Reply:
Wait, it's as simple as asking some other part of the pipeline for the path?

I was under the impression that I'd need to actually create a separate tool that mirrors the functionality of this pipe and then give the hint as to how it should be invoked here.

Reply:
What do you think?

Reply:
Here's the link to what I'm referring to: https://github.com/revng/revng/pull/173/files#diff-8b3d47dd3d3992b5f8d455b306ebe9d3f715dd0c8e6d94b45acdfd4017502640R57-R58.